src/main/java/org/opentripplanner/routing/vertextype/TransitStopVertex.java

```diff
@@ -35,15 +35,21 @@
      */
     private int streetToStopTime = 0;
 
-    public TransitStopVertex (Graph graph, Stop stop) {
+    public TransitStopVertex (Graph graph, Stop stop, TraverseModeSet modes) {
         super(graph, GtfsLibrary.convertIdToString(stop.getId()), stop.getLon(), stop.getLat());
         this.stop = stop;
+        this.modes = modes != null ? modes : new TraverseModeSet();
         this.wheelchairEntrance = stop.getWheelchairBoarding() != WheelChairBoarding.NOT_POSSIBLE;
         isEntrance = false; // Entrance not supported in current otp model
         //Adds this vertex into graph envelope so that we don't need to loop over all vertices
         graph.expandToInclude(stop.getLon(), stop.getLat());
     }
 
+    /** for tests only */
+    public TransitStopVertex (Graph graph, Stop stop) {
```

Comment:
Is there some way to move this into src/test (as a factory method) so it can only be used by tests?